src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/StorePal.cs

```diff
@@ -150,7 +151,9 @@ public static IStorePal FromSystemStore(string storeName, StoreLocation storeLoc
             
             if ((openFlags & OpenFlags.ReadWrite) == OpenFlags.ReadWrite)
             {
-                throw new PlatformNotSupportedException(SR.Cryptography_Unix_X509_MachineStoresReadOnly);
+                throw new CryptographicException(
```

Comment:
Are there any times today where we're throwing a CryptographicException that represents a native error that comes back meaning "not supported on this platform"? e.g. in Sockets if we get back a EAFNOSUPPORT for errno we translate that to a PlatformNotSupportedException.  I'm wondering if in the general "create a CryptographicException to represent this errno" code logic if we need to special case any codes to put a PNSE as an inner exception.